File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map =new HashMap<>();
        for(int i=0;i<nums.length;i++){
            // Integer complements = map.get(nums[i]);
            // if(complements!=null){
            //     return new int[]{i,complements};
            // }
            if(map.containsKey(nums[i])){
                return new int[]{i,map.get(nums[i])};
            }
            map.put(target-nums[i],i);
        }
        return nums;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}
        n = len(nums)

        for i, j in enumerate(nums):
            component = target - j
            if component in numMap:
                return [numMap[component], i]
            numMap[j] = i

#   2,7,11,15